### ISSUE TITLE
GPU usage keeps increasing marginally with each inference request

### ISSUE THREAD

**Issue by shv07** [pytorch/vision#4949]
### 🐛 Describe the bug

I have been trying to deploy the RetinaNet pre-trained model available in torchvision. However, after every inference request with exactly same image, the gpu usage keeps increasing marginally (by roughly 10 MiB, as visible in nvidia-smi). (Same behavior is noticed if I try the same with other Object Detection models like fasterrcnn_resnet50_fpn, fasterrcnn_mobilenet_v3_large_fpn.)

The brief code used for inference
```python

# load the model
self.Model = retinanet_resnet50_fpn(pretrained=True)
self.Model.to(torch.device('cuda'))
self.Model.eval()

# inference fn
@torch.no_grad()
def DetectObjects(self, image, threshold=0.4)->dict:
    image = convert_image_dtype(torch.stack([image]), dtype=torch.float)
    image = image.to(torch.device('cuda'))
    results = self.Model(image)[0]
    gt = results['scores']>threshold
    labels, boxes, scores = results['labels'][gt].cpu().tolist(), results['boxes'][gt].cpu().tolist(), results['scores'][gt].cpu().tolist()
    torch.cuda.empty_cache()
    return boxes, labels, scores
```

### Versions
```
Collecting environment information...
PyTorch version: 1.10.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.10.2
Libc version: glibc-2.17

Python version: 3.7.10 (default, Feb 20 2021, 21:21:24)  [GCC 5.4.0 20160609] (64-bit runtime)
Python platform: Linux-4.15.0-162-generic-x86_64-with-Ubuntu-18.04-bionic
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1660 SUPER
Nvidia driver version: 465.19.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.2.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.2.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.21.4
[pip3] torch==1.10.0
[pip3] torchvision==0.11.1
[conda] Could not collect
```

**Comment by shv07** [comment]
Update: The above behavior disappears if I send large number of request concurrently (say around 5k, leading to OOM for some requests). After doing this, the gpu usage is restored back to exact same value after each inference request is complete. Noticed this behavior in multiple pytorch deployed models.

**Comment by shv07** [comment]
Thanks for your response @fmassa.

Usinng ```torch.cuda.max_memory_allocated()``` indeed deosen't show any residual increase in GPU memory usage. Even the values in ```nvidia_smi``` start showing results as expected but only after a large no. of inference requests have been made.

**Comment by hsonetta** [comment]
`@torch.no_grad()` decorator does not do what it is expected to do i.e. stop gradient updates in inference model. Using `with torch.no_grad()` just before the model inference line stops the gradient updates and won't let the GPU memory grow.
